solr/core/src/test/org/apache/solr/update/processor/UpsertConditionTest.java

```diff
@@ -178,6 +180,87 @@ public void givenNumericField_whenMatching() {
     assertTrue(condition.matches(oldDoc, newDoc));
   }
 
+  @Test
+  public void givenAtomicUpdateSet_whenMatching() {
+    NamedList<String> args = new NamedList<>(ImmutableMap.of(
+        "must", "OLD.field:value",
+        "action", "skip"
+    ));
+
+    UpsertCondition condition = UpsertCondition.parse("skip-it", args);
+
+    assertThat(condition.getName(), is("skip-it"));
+
+    SolrInputDocument oldDoc = new SolrInputDocument();
+    SolrInputDocument newDoc = new SolrInputDocument();
+
+    oldDoc.setField("field", Collections.singletonMap("set", "other1"));
+    assertFalse(condition.matches(oldDoc, newDoc));
+
+    oldDoc.setField("field", Collections.singletonMap("set", "value"));
+    assertTrue(condition.matches(oldDoc, newDoc));
+
+    oldDoc.setField("field", Collections.singletonMap("set", ImmutableList.of("value", "other2")));
+    assertTrue(condition.matches(oldDoc, newDoc));
+
+    oldDoc.setField("field", Collections.singletonMap("set", ImmutableList.of("other1", "other2")));
+    assertFalse(condition.matches(oldDoc, newDoc));
+  }
+
+  @Test
+  public void givenAtomicUpdateAdd_whenMatching() {
+    NamedList<String> args = new NamedList<>(ImmutableMap.of(
+        "must", "OLD.field:value",
+        "action", "skip"
+    ));
+
+    UpsertCondition condition = UpsertCondition.parse("skip-it", args);
+
+    assertThat(condition.getName(), is("skip-it"));
+
+    SolrInputDocument oldDoc = new SolrInputDocument();
+    SolrInputDocument newDoc = new SolrInputDocument();
+
+    oldDoc.setField("field", Collections.singletonMap("add", "other1"));
+    assertFalse(condition.matches(oldDoc, newDoc));
+
+    oldDoc.setField("field", Collections.singletonMap("add", "value"));
+    assertTrue(condition.matches(oldDoc, newDoc));
+
+    oldDoc.setField("field", Collections.singletonMap("add", ImmutableList.of("value", "other2")));
+    assertTrue(condition.matches(oldDoc, newDoc));
+
+    oldDoc.setField("field", Collections.singletonMap("add", ImmutableList.of("other1", "other2")));
+    assertFalse(condition.matches(oldDoc, newDoc));
+  }
+
+  @Test
+  public void givenAtomicUpdateDelete_whenMatching() {
+    NamedList<String> args = new NamedList<>(ImmutableMap.of(
+        "must", "OLD.field:value",
+        "action", "skip"
+    ));
+
+    UpsertCondition condition = UpsertCondition.parse("skip-it", args);
+
+    assertThat(condition.getName(), is("skip-it"));
+
+    SolrInputDocument oldDoc = new SolrInputDocument();
+    SolrInputDocument newDoc = new SolrInputDocument();
+
+    oldDoc.setField("field", Collections.singletonMap("delete", "other1"));
```

Comment:
iirc the verb is `remove` not `delete`.. probably worth renaming to avoid confusion, even though neither are handled, before `remove` is supported later